predict.py

```diff
@@ -11,6 +11,13 @@
 from peft import PeftModel
 import os
 
+PROMPT_TEMPLATE = '''<s>[INST] <<SYS>>
```

Comment:
If we're doing inference with a method that supports conditioning on pre-computed hidden states (e.g. [as Transformers.generate() does](https://github.com/huggingface/transformers/blob/a6484c89b9566333b7f9862e22f702070f47bbe9/src/transformers/models/gpt2/modeling_gpt2.py#L502C1-L503C1)), we could pre-compute and cache the hidden state for the beginning of the prompt and pass it in during prediction.

Reply:
noted! I think this would be excellent but also suspect the work we're doing with vllm/gptq supersedes this for now, but lmk if that's off.